pkg/cluster/spec.go

```diff
@@ -168,6 +168,17 @@ func newWithCliVersion(cliVersion version.Info, opts ...SpecOpt) *Spec {
 	return NewSpec(opts...)
 }
 
+func NewSpecFromClusterConfigWithValidate(clusterConfigPath string, cliVersion version.Info, opts ...SpecOpt) (*Spec, error) {
```

Comment:
another way of naming this
```suggestion
func ReadAndValidateSpecFromClusterConfig(clusterConfigPath string, cliVersion version.Info, opts ...SpecOpt) (*Spec, error) {
```

Reply:
Naming the new cmd pkg function readAndValidateClusterSpec